autoload/ale/sign.vim

```diff
@@ -67,10 +67,14 @@ endfunction
 " Given a loclist, combine the loclist into a list of signs such that only
 " one sign appears per line. Error lines will take precedence.
 " The loclist will have been previously sorted.
-function! ale#sign#CombineSigns(loclist) abort
+function! ale#sign#CombineSigns(buffer, loclist) abort
     let l:signlist = []
 
     for l:obj in a:loclist
+        if has_key(l:obj, 'bufnr') && l:obj.bufnr >= 0 && l:obj.bufnr != a:buffer
```

Comment:
If we were going to allow linters to set warnings or errors for other files, it shouldn't be done without seriously thinking about what the impact of that will be. More likely than not, it will break everything in strange ways.